### PR TITLE
Build Codex from source if repo is included

### PR DESCRIPTION
**Pull request by catrope**
Similar to OOUI, except it's JS only, no PHP.

Also add support for 'main' as an alias for the 'master' branch.
Without this, installing Codex fails, because it doesn't have a
branch named 'master'.

### CODE REVIEW

File: new/postinstall.sh [hunk 1]
```diff
@@ -48,6 +48,17 @@ if [ -d $PATCHDEMO/wikis/$NAME/w/build/ooui ]; then
 	COMPOSER_CACHE_DIR=/dev/null composer require "oojs/oojs-ui @dev" --update-no-dev
 fi
 
+# Codex build
+# TODO test
+if [ -d $PATCHDEMO/wikis/$NAME/w/build/codex ]; then
+	cd $PATCHDEMO/wikis/$NAME/w/build/codex
+	npm install
```
Comment (edg2s) [inline new/postinstall.sh:55]: should be `npm ci`? (And in OOUI too probably)

**Review by edg2s — CHANGES_REQUESTED** [review]

**Comment by edg2s** [comment]
Do you have patches we can test with, and where they would be visible in MW?

**Comment by catrope** [comment]
To test:
- Create a wiki ("A") with change 758961
- Visit wiki A with `?useskin=vector-2022`
- Test the search bar at the top of the page, which uses Codex. When you inspect the search bar, you should see CSS classes beginning with `cdx-`.
-  Create a wiki ("B") with changes 758961 and 765655. This should add Codex to the set of repos automatically, since 765655 is a change in Codex
- Visit wiki B with `?useskin=vector-2022`
- Test the search bar at the top of the page. It should work the same as on wiki A, but you should see different CSS classes. In particular, the `cdx-menu` class should be present on wiki B (on the absolutely positioned container of the typeahead suggestions) and absent on wiki A.

File: new/postinstall.sh (hunk 1 above)
Comment (catrope) [reply new/postinstall.sh:55]: Sure. I just copied the OOUI code, but I'll change both to `npm ci`.

**Comment by edg2s** [comment]
Had to use a different patch as that one is conflicting, but it appears to work: https://patchdemo.wmflabs.org/wikis/405f985b66/wiki/Main_Page?useskin=vector-2022

image

New CSS from that patch:
image

Spent a lot of time in `npm ci`, total build time was 3.5minutes

Also npm had engine warnings - I would recommended only requiring Node 12 in your repos.

**Comment by MatmaRex** [comment]
Nice :o

**Pull request merged**